### PR TITLE
Integrate distributed classification

### PR DESCRIPTION
**Pull request by gavin1332**

### CODE REVIEW

File: python/paddle/fluid/framework.py [hunk 1]
```diff
@@ -691,6 +691,35 @@ def _set_error_clip(self, error_clip):
         """
         self.error_clip = error_clip
 
+    def _set_info(self, key, value):
+        """
+        Set some key-value information associated to this variable.
+
+        Args:
+            key(str): The key of the information entry.
+            value(object): The calue of the information entry.
```
Comment (sandyhouse) [inline python/paddle/fluid/framework.py]: (typo) The **value** of the information entry.

File: python/paddle/fluid/layers/collective.py [hunk 2]
```diff
@@ -178,3 +183,223 @@ def _c_sync_comm_stream(x, ring_id):
...
+        # avoid distributed parameter allreduce gradients
+        weight.is_distributed = True
+        # avoid distributed parameter broadcasting in startup program
+        default_startup_program().global_block().vars[
+            weight.name].is_distributed = True
+
+        bias = None
+        if use_bias:
+            bias = helper.create_parameter(
+                shape=shard_dim, dtype=dtype, is_bias=True)
```
Comment (sandyhouse) [inline python/paddle/fluid/layers/collective.py]: Should shape be in_dim if transpose_weight is True?

File: python/paddle/fluid/framework.py (hunk 1 above)
Comment (gavin1332) [reply python/paddle/fluid/framework.py]: done

File: python/paddle/fluid/layers/collective.py (hunk 2 above)
Comment (gavin1332) [reply python/paddle/fluid/layers/collective.py]: not necessary

**Pull request closed without merging**